instrumentation/crash/src/test/java/io/opentelemetry/android/instrumentation/crash/CrashReportIntegrationTest.kt

```diff
@@ -0,0 +1,233 @@
+    private lateinit var installationContext: InstallationContext
+    private lateinit var instrumentation: CrashReporterInstrumentation
+
+    @Before
+    fun setUp() {
+        exporter = FakeLogRecordExporter()
+        installationContext = fakeInstallationContext(exporter)
+        instrumentation = CrashReporterInstrumentation()
+    }
+
+    @Test
+    fun `test crash reporter instrumentation is installed`() {
+        assertEquals("crash", instrumentation.name)
+        instrumentation.install(installationContext)
+
+        val handler = Thread.getDefaultUncaughtExceptionHandler()
+        assertTrue(handler is CrashReportingExceptionHandler)
+    }
+
+    @Test
+    fun `test uncaught exception captured as log`() {
+        val exc = IllegalStateException("Whoops")
+        val log = simulateUncaughtException(exc)
+
+        log.assertCrashCaptured(
+            expectedStacktrace = exc.stackTraceToString(),
+            thread = Thread.currentThread(),
+            expectedExcType = "java.lang.IllegalStateException",
+            expectedExcMessage = "Whoops",
+        )
+    }
+
+    @Test
+    fun `test exception with no message`() {
+        val exc = IllegalArgumentException()
+        val log = simulateUncaughtException(exc)
+
+        log.assertCrashCaptured(
+            expectedStacktrace = exc.stackTraceToString(),
+            thread = Thread.currentThread(),
+            expectedExcType = "java.lang.IllegalArgumentException",
+        )
+    }
+
+    @Test
+    fun `test java lang error`() {
+        val exc = OutOfMemoryError()
+        val log = simulateUncaughtException(exc)
+
+        log.assertCrashCaptured(
+            expectedStacktrace = exc.stackTraceToString(),
+            thread = Thread.currentThread(),
+            expectedExcType = "java.lang.OutOfMemoryError",
+        )
+    }
+
+    @Test
+    fun `test additional attributes extractors`() {
+        instrumentation.addAttributesExtractor(CustomAttributesExtractor(mapOf("key1" to "value1")))
+        instrumentation.addAttributesExtractor(CustomAttributesExtractor(mapOf("key2" to "value2")))
+
+        val exc = IllegalArgumentException()
+        val log = simulateUncaughtException(exc)
+
+        log.assertCrashCaptured(
+            expectedStacktrace = exc.stackTraceToString(),
+            thread = Thread.currentThread(),
+            expectedExcType = "java.lang.IllegalArgumentException",
+        )
+        val attributes = exporter.awaitLogs(1).single().attributes
+        val attrs = attributes.asMap().mapKeys { it.key.key }
+        assertEquals("value1", attrs["key1"])
+        assertEquals("value2", attrs["key2"])
+    }
+
+    @Test
+    fun `test stackoverflow behavior`() {
+        val exc = generateLargeStacktraceException()
+        val log = simulateUncaughtException(exc)
+        val stacktrace = exc.stackTraceToString()
+
+        // TODO: future: truncate stackframes at some reasonable limit
```

Comment:
We need to truncate stacktraces at some reasonable limit e.g. 500-1000. Production stacktraces can in rare circumstances be much larger than this with repeated frames